napari/_qt/qt_viewer_dock_widget.py

```diff
@@ -76,7 +87,69 @@ def keyPressEvent(self, event):
 
     def _set_title_orientation(self, area):
         if area in (Qt.LeftDockWidgetArea, Qt.RightDockWidgetArea):
-            features = self._initial_features
+            features = self._features
         else:
-            features = self._initial_features | self.DockWidgetVerticalTitleBar
+            features = self._features | self.DockWidgetVerticalTitleBar
         self.setFeatures(features)
+
+
+class QMinimalTitleBar(QLabel):
+    """A widget to be used as the titleBar in the QtMinimalDock Widget.
+
+    Keeps vertical size minimal, has a hand cursor and styles (in stylesheet)
+    for hover.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.setObjectName("QMinimalTitleBar")
+
+        layout = QHBoxLayout()
+        layout.setContentsMargins(8, 0, 8, 0)
+        layout.setSpacing(4)
+
+        line = QFrame(self)
+        line.setFixedHeight(1)
+        line.setObjectName("QMinimalTitleBarLine")
+
+        self.close_button = QPushButton(self)
+        self.close_button.setCursor(Qt.ArrowCursor)
+
+        layout.addWidget(self.close_button)
+        layout.addWidget(line)
+
+        self.setLayout(layout)
+        self.setCursor(Qt.OpenHandCursor)
+
+    def sizeHint(self):
+        # this seems to be the correct way to set the height of the titlebar
+        szh = super().sizeHint()
+        szh.setHeight(18)
+        return szh
+
+
+class QtMinimalDockWidget(QtViewerDockWidget):
```

Comment:
I might call it `QtMinimalDockWidget` - though actually what does it look like it we just always use this one, i.e. even for the console. I might prefer it

Reply:
would simplify the code base and uniformity of stylings just to have one `QtViewerDockWidget` which had those minimal styles (either with title bar horizontal or vertical)

Reply:
would also remove the re-docking after close problem for all our added widgets now right? apologies if you already planned to just replace `QtViewerDockWidget` and I'm slow to catch on

Reply:
well I hadn't necessarily... but I think it probably makes sense.  I don't mind losing that little "float" icon.  The functionality is triply duplicated, since you can already tear it out with a click and drag, and a double click on the title bar does what the float icon did.  The one thing I don't always love is the non-native window look on floated windows.  I wish we could have it both ways... but the native title bar brings back the re-docking problem.  (i even tried to intercept and ignore the `closeEvent`, but couldn't get it to work).  But let me work on it a little more from the perspective of making it the global DockWidget style

Reply:
Losing that undock button is good by me, as you say very redundant functionality, especially now that the cursor turns to a hand when you hover. I don’t mind losing the native titlebar - I find the switch to a gray styling very off putting in the Mac. In fact after this I might want us to make a follow up PR where we drop the native titlebar on the main viewer window too, but that can come later

Reply:
>I might want us to make a follow up PR where we drop the native titlebar on the main viewer window too, but that can come later

![giphy](https://user-images.githubusercontent.com/1609449/69924559-1a6ae480-147a-11ea-9e43-4ff6c2490761.gif)

Reply:
I strongly oppose getting rid of the native titlebar. I actually quite liked the mini titlebars on the undocked windows, but I'm ok with them being special — most people expect that and most people wouldn't want to e.g. maximise them or move them to the dock. For the main window, not so much. I think it's very important that napari looks and acts like any other application in the user's system. The familiarity is super important. Everyone knows that pushing the little green button on napari makes it full-screen, (most people know alt-pushing it maximises it), pushing the yellow button puts in it the dock, etc. This is super important.

Reply:
We can also make the title bar try and obey the appropriate conventions with the different color buttons while styling it ourselves. Chrome and Illustrator both do that for me - i.e. illustrator looks as follows 
![image](https://user-images.githubusercontent.com/6531703/69976670-386a3080-14de-11ea-9d4d-5fddfeb4fdea.png)
and chrome dark mode looks similar. I'd just want to do something similar, but we can discuss this in its own issue